core/src/main/java/org/bitcoinj/base/BitcoinNetwork.java

```diff
@@ -84,21 +86,33 @@ public Coin maxMoney() {
      * @throws IllegalArgumentException if there is no matching enum
      */
     public static BitcoinNetwork of(String nameString) {
-        return find(nameString)
+        return findCanonicalName(nameString)
                 .orElseThrow(() -> new IllegalArgumentException("Unrecognized network name : " + nameString));
     }
 
     /**
-     * Find the {@code BitcoinNetwork} from a name String
+     * Find the {@code BitcoinNetwork} from a name string, or from a number of common alternate names
      * @param nameString A name string (e.g. "MAIN", "TEST", "SIGNET")
      * @return An {@code Optional} containing the matching enum or empty
      */
     public static Optional<BitcoinNetwork> find(String nameString) {
+        final Optional<BitcoinNetwork> canonicalName = findCanonicalName(nameString);
```

Comment:
This logic deserves a second look by someone used to Optionals. I figured the `.or()` method might be suitable here, but it's only available on Java9+. Converting the two Optionals into a stream and using `filter()/findFirst()` seemed like overkill to me.

Reply:
Yeah, `or()` is what we really want here. I think the best we can do is:
```
        return canonicalName.isPresent()
                ? canonicalName
                : findAlternateName(nameString);
```

Reply:
(Java 9 -- and to a lesser extent 10 and 11 -- really smoothed some of the rough edges in the new features of Java 8.)

Reply:
Thanks! I've pushed the change.